gmi_egads/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+if(DEFINED TRIBITS_PACKAGE)
+  include(pkg_tribits.cmake)
+  return()
+endif()
+
+if(NOT ENABLE_EGADS)
+  return()
+endif()
+
+# this file brings PUMI_USE_EGADSLITE from CMake to C++
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/gmi_egads_config.h.in"
+               "${CMAKE_CURRENT_BINARY_DIR}/gmi_egads_config.h")
```

Comment:
@jacobmerson this config file only defines `PUMI_USE_EGADSLITE`. Is conditionally calling `target_compile_definitions` a better approach here? It seems like it would be to me.

Reply:
`target_compile_definitions` would be used instead of https://github.com/tuckerbabcock/core/blob/afb21d570ad03e75c346b4a45b714a0be1dbb815/CMakeLists.txt#L103

An example of its use is here:

https://github.com/SCOREC/core/blob/58a8832c9b8306fc62312ff8234479a05eeecba0/CMakeLists.txt#L159

Reply:
@cwsmith would you prefer the target be the interface target `core`, or just the `gmi_egads` target? I think the `gmi_egads` target would be more explicit and therefore preferable, but I don't have strong thoughts. 

For my above thought on having `PUMI_USE_EGADSLITE` be set with `target_compile_definitions`, I was imagining it would be set as a `PRIVATE` property of `gmi_egads`, since I think that would be semantically equivalent to the current setup with the config file.

Reply:
I spent some more time looking at the code and have a hopefully more complete response.  

IIUC, the user needs to know if PUMI was installed with EGADS and/or EGADSLite support and `gmi_egads/gmi_egads.c` needs to know if EGADSLite is enabled (`gmi_egads.c` is only compiled if egads is enabled).  If that is correct, I think the best option is to use `target_compile_definitions(...)` for both variables and remove the config header.  We would also want to replace any use of `HAVE_EGADS[LITE]` with the `PUMI_HAS_EGADS[LITE]` variable (or equivalent).  

To answer your questions:

- target:  Try setting it on `gmi_egads`, but that may cause problems compiling tests that use the preprocessor variables when gmi_egads is not available/installed.
- visibility: If the first assumption was correct, then I think they should both have `INTERFACE` scoping/visibility.

Reply:
@cwsmith I've switched over all the EGADS stuff to use `target_compile_definitions`, and renamed the flags from `HAVE_EGADS` to `PUMI_HAS_EGADS`. `PUMI_HAS_EGADS` is a private definition on `gmi_egads` since it is only needed in `gmi_egads.c`, while `PUMI_HAS_EGADS` is an interface definition on `gmi_egads` since its needed by people who use the library. I did not experience any issues compiling and running the tests that use the these variables when gmi_egads wasn't built, so I think it should be okay.